ca/django_ca/managers.py

```diff
@@ -336,9 +342,21 @@ def init(
         # NOTE: Already verified by KeySizeAction, so these checks are only for when the Python API is used
         #       directly.
         generate_key_args = validate_key_parameters(key_size, key_type, ecc_curve)
-        algorithm = parse_hash_algorithm(algorithm)
+        if openssh_ca:
+            algorithm = parse_hash_algorithm(algorithm)
+        else:
+            algorithm = None
```

Comment:
I think that's a bug. This will force the algorithm to `None` if this is _not_ an OpenSSH CA. That's the inverse of what was there.

Reply:
Gah nice catch! :man_facepalming:  Fixed now.